systemtests/tests/py3plug-fd-libcloud/testrunner-default

```diff
@@ -64,49 +60,74 @@ rm ${tmp}/data/weird-files/newline*
 rm ${tmp}/data/weird-files/tab*
 # the following file also makes problems
 rm ${tmp}/data/weird-files/filename-with-non-utf8-bytestring*
-# s3cmd does not sync empty dirs
+
+# s3cmd does not sync empty dirs, but you can create them by placing
+# into them an empty dummy file with same name
+# Most end-user expect to be able to restore those empty dirs.
 rmdir ${tmp}/data/weird-files/big-X
 rmdir ${tmp}/data/weird-files/subdir
+mkdir -p ${tmp}/data/weird-files/empty_subdir/
+touch ${tmp}/data/weird-files/empty_subdir/empty_subdir
+${S3} sync "${BackupDirectory}" s3://${bucket_name}
+# create directly an empty directory in the s3
+# unfortunately doesn't work
+# ${S3} put remote_empty s3://${bucket_name}/remote_empty/
 
-${S3} sync "$BackupDirectory" s3://$bucket_name
+echo "Data preparation done"
 
 start_test
 
-cat <<END_OF_DATA >$tmp/bconcmds
+
+cat <<END_OF_DATA >${tmp}/bconcmds
 @$out /dev/null
 messages
-@$out $tmp/log1.out
-setdebug level=100 storage=File
-setdebug level=200 client=bareos-fd trace=1 timestamp=1
-label volume=TestVolume001 storage=File pool=Full
-run job=$JobName yes
+@$out ${tmp}/log1.out
+setdebug level=100 trace=1 timestamp=1 storage=File
+setdebug level=200 trace=1 timestamp=1 client=bareos-fd
+run job=${JobName} yes
+wait
+setdebug level=0 client=bareos-fd
 status director
 status client
 status storage=File
 wait
 messages
+END_OF_DATA
+
+echo "Full backup stage"
+run_bconsole
+expect_grep "Backup OK" "${tmp}/log1.out" "Full Backup not found!"
+if [ ${estat} -ne 0 ]; then
+    exit ${estat}
+fi
+
+cat <<END_OF_DATA >${tmp}/bconcmds
 @#
 @# now do a restore
 @#
-@$out $tmp/log2.out
-wait
-restore client=bareos-fd fileset=PluginTest where=$tmp/bareos-restores select all done
-yes
+@$out ${tmp}/log2.out
+setdebug level=200 trace=1 timestamp=1 client=bareos-fd
+restore client=bareos-fd fileset=PluginTest where=${tmp}/bareos-restores select all done yes
 wait
 messages
 quit
 END_OF_DATA
+echo "Restore backup stage"
+run_bconsole
+expect_grep "Restore OK" "${tmp}/log2.out" "Restore Backup not ok!"
+if [ ${estat} -ne 0 ]; then
+    exit ${estat}
+fi
 
-run_bareos "$@"
 check_for_zombie_jobs storage=File
 
 check_two_logs
 list=( $(find "${BackupDirectory}" -type f) )
 # Using check_restore_only_files_diff instead of check_restore_diff
-# to don'"t diff the file attributes, because they are not saved
-#check_restore_only_files_diff "${list[@]}"
+# to don't diff the file attributes, because they are not saved
```

Comment:
```suggestion
# to not diff the file attributes, because they are not saved
```

Alltough you can probably remove this completely since we do not use `check_restore_only_files_diff` anymore and are instead relying on diff directly (see below).